source/upgrade.txt

```diff
@@ -0,0 +1,173 @@
+versions. For example, if you are upgrading the driver from v3.x to
+v5.x, address all breaking changes listed under v4.x and v5.x.
+
+.. _node-breaking-changes-v5.x:
+
+Version 5.x Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Driver versions 5.x are not compatible with Node.js
+  v12 or earlier. If you want to use this version of the driver, You must
+  use Node.js v14.20.1 or greater.
+
+- The driver removes support for callbacks in favor of a promise-based API.
+  The following list provides some strategies for callback users to adopt this
+  version:
+
+  - Migrate to the promise-based API (recommended)
+  - Use the promise-based API and ``util.callbackify``
+  - Add ``mongodb-legacy`` to continue using callbacks
+
+For more information about these strategies, see
+`the v5.0 changelog <https://github.com/mongodb/node-mongodb-native/blob/main/etc/notes/CHANGES_5.0.0.md#optional-callback-support-migrated-to-mongodb-legacy>`__.
+
+- The driver removes support for the ``Collection.insert()``,
+  ``Collection.update()``, and ``Collection.remove()`` helper methods.
+  The following list provides instructions on how to replace the
+  functionality of the removed methods:
+
+  - Migrate from ``Collection.insert()`` to ``insertOne()`` or ``insertMany()``
+  - Migrate from ``Collection.update()`` to ``updateOne()`` or ``updateMany()``
+  - Migrate from ``Collection.remove()`` to ``deleteOne()`` or ``deleteMany()``
+
+- The driver no longer includes AWS SDK modules by default.
+
+- The driver no longer automatically imports the ``bson-ext`` package.
+
+- The driver removes support for custom ``Promise`` libraries. The driver no
+  longer supports the ``promiseLibrary`` option of ``MongoClient`` and the ``Promise.set``
+  export that allows specifying a custom ``Promise`` library.
+
+- The driver removes support for the ``Collection.mapReduce()`` helper.
+
+- The ``BulkWriteResult`` type no longer has the publicly enumerable
+  ``result`` property.
+
+- The following types, options, and methods have been removed:
+
+  - ``BulkResult.lastOp()`` method
+  - ``opTime`` property of ``BulkResult``
+  - ``BulkWriteOptions.keepGoing`` option
+  - ``WriteConcernError.err()`` method
+  - ``AddUserOptions.digestPassword`` option
+  - Kerberos ``gssapiCanonicalizeHostName`` option
+  - ``slaveOk`` options and method removed in favor of ``secondaryOk``
+  - ``ObjectID`` type removed in favor of ``ObjectId``
+  - ``AsyncIterator`` interface removed in favor of ``AsyncGenerator``
+
+.. _node-breaking-changes-v4.x:
+
+Version 4.x Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Driver versions 4.x are not compatible with Node.js
+  v12.8 or earlier. If you want to use this version of the driver, You must
+  use Node.js v12.9 or greater.
+
+- ``Cursor`` types no longer extend ``Readable`` directly.
+
+- You cannot use a ``ChangeStream`` instance as an iterator after using
+  it as an ``EventEmitter`` nor visa versa.
```

Comment:
Suggestion: I think "vice versa" is the correct term here. However, I'm not sure how I feel about using it in our documentation. I don't know if the meaning of it is universally understood? I could be off-base on that assumption though so I'll defer to you on whether we keep it in or not!

Reply:
I rephrased, and I don't think I violated the style guide, but lmk